src/shell/commands/disk_rebalance.cpp

```diff
@@ -0,0 +1,280 @@
+{
+    std::map<dsn::rpc_address, dsn::replication::node_status::type> nodes;
+    auto error = sc->ddl_client->list_nodes(::dsn::replication::node_status::NS_INVALID, nodes);
+    if (error != dsn::ERR_OK) {
+        fmt::print(stderr, "list nodes failed, error={}\n", error.to_string());
+        return false;
+    }
+
+    std::vector<dsn::rpc_address> targets;
+    if (!node_address.empty()) {
+        for (const auto &node : nodes) {
+            if (node.first.to_std_string() == node_address) {
+                targets.emplace_back(node.first);
+            }
+        }
+        if (targets.empty()) {
+            fmt::print(stderr, "please input valid target node_address!\n");
+            return false;
+        }
+    } else {
+        for (const auto &node : nodes) {
+            targets.emplace_back(node.first);
+        }
+    }
+
+    sc->ddl_client->query_disk_info(targets, err_resps, app_name);
+    return true;
+}
+
+bool query_disk_capacity(command_executor *e, shell_context *sc, arguments args)
+{
+    // disk_capacity [-n|--node replica_server(ip:port)] [-o|--out file_name][-j|-json][-d|--detail]
+    const std::vector<std::string> &params = {"n", "node", "o", "out"};
+    const std::vector<std::string> &flags = {"j", "json", "d", "detail"};
+    argh::parser cmd(args.argc, args.argv, argh::parser::PREFER_PARAM_FOR_UNREG_OPTION);
+    if (!parse_cmd(cmd, params, flags)) {
+        return false;
+    }
+
+    bool format_to_json = cmd[{"-j", "--json"}];
+    bool query_detail_info = cmd[{"-d", "--detail"}];
+    std::string node_address = cmd({"-n", "--node"}).str();
+    std::string file_name = cmd({"-o", "--out"}).str();
+
+    std::map<dsn::rpc_address, dsn::error_with<query_disk_info_response>> err_resps;
+    if (!query_disk_info(sc, cmd, node_address, err_resps)) {
+        return false;
+    }
+
+    std::ostream *ostream_ptr = get_out_stream(file_name);
+    if (!ostream_ptr) {
+        fmt::print(stderr, "get output stream failed!");
+        return false;
+    }
+    std::ostream &out = *ostream_ptr;
+
+    dsn::utils::table_printer node_printer;
+    node_printer.add_title("node");
+    node_printer.add_column("total_capacity(MB)");
+    node_printer.add_column("avalable_capacity(MB)");
+    node_printer.add_column("avalable_ratio(%)");
+    node_printer.add_column("capacity_balance");
+
+    dsn::utils::multi_table_printer multi_printer;
+    for (const auto &err_resp : err_resps) {
+        dsn::error_s err = err_resp.second.get_error();
+        if (err.is_ok()) {
+            err = dsn::error_s::make(err_resp.second.get_value().err);
+        }
+        if (!err.is_ok()) {
+            fmt::print(stderr,
+                       "disk of node[{}] info skiped because request failed, error={}\n",
+                       err_resp.first.to_std_string(),
+                       err.description());
+            if (!node_address.empty()) {
+                return false;
+            }
+        } else {
```

Comment:
Add `continue` in `if` block, then remove `else`, try to make code simple and clear.